Lint job failed on head

/area dev-productivity
/kind technical-debt

**What happened**:
After merging this PR:  https://github.com/gardener/etcd-backup-restore/pull/917 `make verify` is failing on head job https://github.com/gardener/etcd-backup-restore/actions/runs/20327164884/job/58394682955, although it passed on the PR 🤷 

Lint error:
```
Executing golangci-lint
Error: pkg/types/snapstore.go:266:6: exported: exported type SecondarySnapstoreConfig should have comment or be unexported (revive)
type SecondarySnapstoreConfig struct {
     ^
Error: pkg/types/snapstore.go:272:1: exported: exported method SecondarySnapstoreConfig.AddFlags should have comment or be unexported (revive)
func (c *SecondarySnapstoreConfig) AddFlags(fs *flag.FlagSet) {
^
Error: pkg/types/snapstore.go:278:1: exported: exported method SecondarySnapstoreConfig.Validate should have comment or be unexported (revive)
func (c *SecondarySnapstoreConfig) Validate() error {
^
Error: pkg/types/snapstore.go:285:1: exported: exported method SecondarySnapstoreConfig.Complete should have comment or be unexported (revive)
func (c *SecondarySnapstoreConfig) Complete() {
^
Error: pkg/snapstore/init.go:21:1: exported: exported function NewSecondarySnapstoreConfig should have comment or be unexported (revive)
func NewSecondarySnapstoreConfig() *brtypes.SecondarySnapstoreConfig {
^
Error: pkg/snapshot/copier/copier.go:254:1: exported: exported method Copier.SyncBackups should have comment or be unexported (revive)
func (c *Copier) SyncBackups(ctx context.Context, interval time.Duration) error {
^
Error: pkg/types/snapstore.go:190:22: fieldalignment: struct with 96 pointer bytes could be 72 (govet)
type SnapstoreConfig struct {
                     ^
Error: pkg/snapshot/copier/copier.go:37:13: fieldalignment: struct of size 104 could be 96 (govet)
type Copier struct {
            ^
Error: pkg/server/backuprestoreserver.go:249:6: shadow: declaration of "cp" shadows declaration at line 196 (govet)
					cp := copier.NewCopier(b.logger, ss, secondary, -1, -1, 10, false, 0)
```

**What you expected to happen**:


**Anything else we need to know?**:

--- a/pkg/server/backuprestoreserver.go
+++ b/pkg/server/backuprestoreserver.go
@@ -246,8 +246,8 @@ func (b *BackupRestoreServer) runServer(ctx context.Context, restoreOpts *brtype
 						b.logger.Fatalf("failed to create snapshot backup copier: %v", err)
 					}
 
-					cp := copier.NewCopier(b.logger, ss, secondary, -1, -1, 10, false, 0)
-					if err := cp.SyncBackups(ctx, b.config.SecondarySnapstoreConfig.SyncPeriod.Duration); err != nil {
+					cp = copier.NewCopier(b.logger, ss, secondary, -1, -1, 10, false, 0)
+					if err := cp.SyncBackups(leCtx, b.config.SecondarySnapstoreConfig.SyncPeriod.Duration); err != nil {
 						b.logger.Fatalf("failed to sync backups: %v", err)
 					}
 					go backupssr.RunGarbageCollector(backupGcStop)
@@ -274,10 +274,6 @@ func (b *BackupRestoreServer) runServer(ctx context.Context, restoreOpts *brtype
 		OnStoppedLeading: func() {
 			if runServerWithSnapshotter {
 				b.logger.Info("backup-restore stops leading...")
-				if cp != nil {
-					b.logger.Infof("Stopping backup copier.")
-					cp.Stop()
-				}
 				close(backupGcStop)
 				handler.SetSnapshotterToNil()
 

--- a/pkg/snapshot/copier/copier.go
+++ b/pkg/snapshot/copier/copier.go
@@ -41,11 +41,10 @@ type Copier struct {
 	maxBackups                  int
 	maxBackupAge                int
 	maxParallelCopyOperations   int
-	waitForFinalSnapshot        bool
 	waitForFinalSnapshotTimeout time.Duration
 	mu                          sync.Mutex
 	running                     bool
-	stopCh                      chan struct{}
+	waitForFinalSnapshot        bool
 }
 
 // NewCopier creates a new copier.
@@ -251,6 +250,7 @@ func (c *Copier) doWaitForFinalSnapshot(ctx context.Context, interval time.Durat
 	}
 }
 
+// SyncBackups periodically synchronizes backups to the secondary snapstore.
 func (c *Copier) SyncBackups(ctx context.Context, interval time.Duration) error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
@@ -287,20 +287,6 @@ func (c *Copier) syncBackups(ctx context.Context, interval time.Duration) {
 			if err := c.CopyBackups(ctx); err != nil {
 				c.logger.Errorf("could not copy backups: %v", err)
 			}
-		case <-c.stopCh:
-			c.logger.Info("Backup copier is shutting down")
-			return
 		}
 	}
-}
-
-// Stop stops the background copier goroutine
-func (c *Copier) Stop() {
-	c.mu.Lock()
-	defer c.mu.Unlock()
-	if !c.running {
-		return
-	}
-	close(c.stopCh)
-	c.running = false
 }

--- a/pkg/snapstore/init.go
+++ b/pkg/snapstore/init.go
@@ -18,6 +18,7 @@ func NewSnapstoreConfig() *brtypes.SnapstoreConfig {
 	}
 }
 
+// NewSecondarySnapstoreConfig returns the secondary-snapstore config.
 func NewSecondarySnapstoreConfig() *brtypes.SecondarySnapstoreConfig {
 	return &brtypes.SecondarySnapstoreConfig{
 		StoreConfig:       NewSnapstoreConfig(),

--- a/pkg/types/snapstore.go
+++ b/pkg/types/snapstore.go
@@ -195,6 +195,9 @@ type SnapstoreConfig struct {
 	Prefix string `json:"prefix,omitempty"`
 	// Temporary Directory
 	TempDir string `json:"tempDir,omitempty"`
+	// EnvPrefix is the prefix to be used for environment variables.
+	// It is used to differentiate between primary and secondary snapstore configs.
+	EnvPrefix string `json:"envPrefix,omitempty"`
 	// MaxParallelChunkUploads holds the maximum number of parallel chunk uploads allowed.
 	MaxParallelChunkUploads uint `json:"maxParallelChunkUploads,omitempty"`
 	// MinChunkSize holds the minimum size for a multi-part chunk upload.
@@ -203,9 +206,6 @@ type SnapstoreConfig struct {
 	IsSource bool `json:"isSource,omitempty"`
 	// IsEmulatorEnabled indicates whether a storage emulator is being used for the snapstore.
 	IsEmulatorEnabled bool `json:"isEmulatorEnabled,omitempty"`
-	// EnvPrefix is the prefix to be used for environment variables.
-	// It is used to differentiate between primary and secondary snapstore configs.
-	EnvPrefix string `json:"envPrefix,omitempty"`
 }
 
 // AddFlags adds the flags to flagset.
@@ -262,18 +262,21 @@ func (c *SnapstoreConfig) MergeWith(other *SnapstoreConfig) {
 	}
 }
 
+// SecondarySnapstoreConfig defines the configuration to enable and create secondary snapshot store.
 type SecondarySnapstoreConfig struct {
 	StoreConfig       *SnapstoreConfig
 	BackupSyncEnabled bool              `json:"backupSyncEnabled,omitempty"`
 	SyncPeriod        wrappers.Duration `json:"syncPeriod,omitempty"`
 }
 
+// AddFlags adds the flags to flagset and also adds `secondary-` prefix for all snapstore parameters.
 func (c *SecondarySnapstoreConfig) AddFlags(fs *flag.FlagSet) {
 	c.StoreConfig.addFlags(fs, "secondary-")
 	fs.BoolVar(&c.BackupSyncEnabled, "secondary-backup-sync-enabled", c.BackupSyncEnabled, "enable secondary backup-sync feature")
 	fs.DurationVar(&c.SyncPeriod.Duration, "secondary-backup-sync-period", c.SyncPeriod.Duration, "period for periodic backup sync operations")
 }
 
+// Validate validates the config.
 func (c *SecondarySnapstoreConfig) Validate() error {
 	if c.BackupSyncEnabled {
 		return c.StoreConfig.Validate()
@@ -281,6 +284,7 @@ func (c *SecondarySnapstoreConfig) Validate() error {
 	return nil
 }
 
+// Complete completes the config.
 func (c *SecondarySnapstoreConfig) Complete() {
 	c.StoreConfig.EnvPrefix = "SECONDARY_"
 	c.StoreConfig.Complete()

